Vagrantfile.fedora

```diff
@@ -17,8 +17,14 @@ Vagrant.configure("2") do |config|
     # Work around dnf mirror failures by retrying a few times
     for i in $(seq 0 2); do
       sleep $i
-      cat << EOF | dnf -y shell && break
-config exclude kernel,kernel-core
+      # 1. "config exclude" dnf shell command is not working in Fedora 35
+      # (see https://bugzilla.redhat.com/show_bug.cgi?id=2022571);
+      # the workaround is to specify it as an option.
+      # 2. systemd 249.6-2.fc35 has a bug preventing rootless containers
+      # from starting when --systemd-cgroup is used for runc run
+      # (see https://bugzilla.redhat.com/show_bug.cgi?id=2022041),
+      # the workaround is not to upgrade systemd.
+      cat << EOF | dnf -y --exclude=kernel,kernel-core,systemd,systemd-* shell && break
```

Comment:
(OT) I'm curious; what's the advantage of using `dnf shell` in a script over just `dnf update, dnf install..` ? (IIUC, `dnf shell` is more for interactive use?)

Reply:
`dnf shell` is for transactional use. IOW, if you want to, say, install, remove, and update something in a single transaction (rather than 3 separate ones), `dnf shell` is your friend. Surely, it can be done in 3 transactions, but 1 is slightly faster as dnf does some things (reading rpmdb, calculating dependencies, writing rpmdb, and somesuch) only once rather than thrice (or twice, in the code we discuss).

Reply:
Ah, right, thanks for explaining; it felt a bit odd